it/uk/gov/hmrc/agentrelationships/controllers/RelationshipsControllerISpec.scala

```diff
@@ -483,124 +479,125 @@ class RelationshipISpec extends UnitSpec
 
     //UNHAPPY PATHS
 
-    "return 404 when credentials are not found in gg" in {
-      givenAgentCredentialsAreNotFoundFor(Arn(arn))
-      givenAgentCodeIsFoundFor("foo", "bar")
+    "return 404 when credentials are not found in es" in {
+      givenPrincipalGroupIdNotExistsFor(arn)
+      givenGroupInfo("foo", "bar")
       givenAgentIsAllocatedAndAssignedToClient(vrn, "bar")
-      givenAuditConnector()
+      givenDelegatedGroupIdsNotExistForKey(s"HMCE-VATDEC-ORG~VATRegNo~${vrn.value}")
+
       val result = await(doRequest)
       result.status shouldBe 404
-      (result.json \ "code").as[String] shouldBe "INVALID_ARN"
+      (result.json \ "code").as[String] shouldBe "RELATIONSHIP_NOT_FOUND"
     }
 
-    "return 404 when agent code is not found in gg" in {
-      givenAgentCredentialsAreFoundFor(Arn(arn), "foo")
-      givenAgentCodeIsNotInTheResponseFor("foo")
-      givenAgentIsAllocatedAndAssignedToClient(vrn, "bar")
-      givenAuditConnector()
+    "return 404 when agent code is not found in ugs" in {
+      givenPrincipalUser(arn, "foo")
+      givenGroupInfoNotExists("foo")
+      givenDelegatedGroupIdsExistFor(vrn, Set("foo"))
+      givenDelegatedGroupIdsNotExistForKey(s"HMCE-VATDEC-ORG~VATRegNo~${vrn.value}")
+
       val result = await(doRequest)
       result.status shouldBe 404
-      (result.json \ "code").as[String] shouldBe "UNKNOWN_AGENT_CODE"
+      (result.json \ "code").as[String] shouldBe "RELATIONSHIP_NOT_FOUND"
     }
 
     //FAILURE CASES
 
-    "return 502 when GsoAdminGetCredentialsForDirectEnrolments returns 5xx" in {
-      whenGetCredentialsReturns(500)
-      givenAgentCodeIsFoundFor("foo", "bar")
+    "return 502 when ES1/principal returns 5xx" in {
+      givenPrincipalGroupIdRequestFailsWith(500)
+      givenGroupInfo("foo", "bar")
       givenAgentIsAllocatedAndAssignedToClient(vrn, "bar")
-      givenAuditConnector()
+
       val result = await(doRequest)
       result.status shouldBe 502
     }
 
-    "return 502 when GsoAdminGetUserDetails returns 5xx" in {
-      givenAgentCredentialsAreFoundFor(Arn(arn), "foo")
-      whenGetUserDetailReturns(500)
+    "return 502 when UGS returns 5xx" in {
+      givenPrincipalUser(arn, "foo")
+      givenGroupInfoFailsWith(500)
```

Comment:
`givenGroupInfoFailsWith(500)` potentially conflicts with the
`givenGroupInfo("foo", "bar")` stub called in `givenAgentIsAllocatedAndAssignedToClient(_, "bar")`